guides/embedding.md

```diff
@@ -0,0 +1,227 @@
+
+const h2 = new Hydra({ makeGlobal: false, detectAudio: false }).synth
+h2.shape(4).diff(h2.osc(2, 0.1, 1.2)).out()
+```
+
+See https://glitch.com/edit/#!/multi-hydra for a working example of multiple hydra canvases, created by Naoto Hieda.
+
+##### Make global a select group of functions
+
+To keep the same syntax as hydra in non-global mode, consider destructuring the object further. This can also allow to expose only a handful of the functions in Hydra.
+```javascript
+const { src, osc, gradient, shape, voronoi, noise, s0, s1, s2, s3, o0, o1, o2, o3, render } = hydra
+shape(4).diff(osc(2, 0.1, 1.2)).out()
+```
+
+## Hydra as a website's background
+
+### HTML
+
+```html
+<body>
+	<canvas id="hydra-bg"></canvas>
+	<div class="content">
+		<h1>Title</h1>
+		Some text :)
+	</div>
+</body>
+```
+
+### CSS
+
+```css
+#hydra-bg {
+	position: fixed; /* ignore margins */
+	top: 0px;
+	left: 0px;
+	width: 100%; /* fill screen */
+	height: 100%;
+	background-size: cover;
+	overflow-y: hidden;
+	z-index: -1; /* place behind everything else */
+	display: block;
+}
+
+.content {
+	background-color: rgba(0,0,0,0.3);
+	margin: 6%;
+}
+```
+
+Changing the z-index to a very high value, the size of the canvas to something small (make sure to change it in JS as well), and the pixel position to wherever you desire, would effectively change the website to one where Hydra is an overlay.
+
+### JS
+
+```javascript
+let hydraCanvas = document.getElementById("hydra-bg");
+// set small size to avoid high resource demand:
+hydraCanvas.width  = Math.min(window.innerWidth  / 2, 1280);
+hydraCanvas.height = Math.min(window.innerHeight / 2, 720);
+
+const hydra = new Hydra({
+  canvas: hydraCanvas,
+  detectAudio: false,
+  enableStreamCapture: false,
+});
+
+osc().blend(noise()).out();
+```
+
+#### Note on scrolling
+
+You may find useful using `window.scrollY` (or X) inside your background's Hydra sketch for it to change as you scroll. The event listener `onscroll` might also be helpful.
+
+#### Note on resize
+
+You may want to automatically resize the canvas' size as the user resizes the window:
+
+```javascript
+resizeTimeout = -1;
+onresize = ()=> {
+	clearTimeout(resizeTimeout);
+    resizeTimeout = setTimeout(()=>{
+		setResolution(
+			Math.min(window.innerWidth  / 2, 1280),
+			Math.min(window.innerHeight / 2, 720)
+		);
+	},200)
+}
+// using timeouts to avoid setting the resolution constantly while the user is resizing 
+```
+
+### Live example
+
+You can find a live example [here](https://hydra-background-webpage.glitch.me/). You can open the glitch editor for it [here](https://glitch.com/edit/#!/hydra-background-webpage).
+
+## Scrollable Hydra webpage with different codeblocks
+
+Given a webpage with a Hydra background (just like the described above), and a series of code blocks to be evaluated across the website's scroll size, you can create a new IntersectionObserver for each codeblock so that its code is run whenever the user scrolls to it. For example:
```

Comment:
i think it's useful because many users sometimes want to make sort of a gallery of hydra code and this helps. but this section definitely needs better wording heh